### PR TITLE
Support for event_members matching in matching callbacks

### PR DESCRIPTION
**Pull request by DonAurelio**
* Now, in addition to `event_name`, matching rules will consider also`event_members`.
* One `member_regex` **MUST** match one and only one `event_member`.
* One `event_member` **CAN BE** matched by only one `member_regex`.

### CODE REVIEW

**Comment by DonAurelio** [comment]
I am ready for your comments Thomas, I give up with the `eval` even when quoted strings are present in the YAML it parse it as a variable. I think this is because the YAML load do not scape the quoates.

File: bin/metababel [hunk 1]
```diff
@@ -23,6 +23,31 @@ def sanitize(str)
...
+  match ? str.match?(match['expr']) : str == pattern_or_str
+end
+
+def btx_event_match(evt_name, evt_args, match_name, match_args)
+  # Return nil if the event and args do not match.
+  # Return a list otherwise.
+
+  # If match_args.length > evt_args.length, a match_arg must match one evt_arg
+  # twice, but that is not allowed. 
+  if btx_match?(evt_name, match_name) and evt_args.length >= match_args.length
```
Comment (TApplencourt) [inline bin/metababel:38]: Wrong tests,
You want to tests the `matched` and not the `match`.

File: bin/metababel [hunk 2]
```diff
@@ -23,6 +23,31 @@ def sanitize(str)
...
+    matched_args = match_args.filter_map do |match_arg|
+      matched_event_args = evt_args.find_all { |evt_arg| btx_match?(evt_arg.name, match_arg) }
+    end.flatten
+
+    unless matched_args.uniq.length == match_args.length
+      raise "Memeber regex matched more than one event member, got '#{matched_args.map(&:name).join(', ') }' from '#{match_args.join(', ') }'."
+    end
```
Comment (TApplencourt) [inline bin/metababel:45]: You need to check also that `matched_args.uniq.length == matched_args.length`, so you are sure you don't match the same twice

File: bin/metababel [hunk 3]
```diff
@@ -23,6 +23,31 @@ def sanitize(str)
   str.gsub(/[^0-9A-Za-z-]/, '_')
 end
 
+def btx_match?(str, pattern_or_str)
+  # String beginning and ending in '/' are treated as regexp.
+  match = pattern_or_str.match(%r{^/(?<expr>.+)/$})
+  match ? str.match?(match['expr']) : str == pattern_or_str
```
Comment (TApplencourt) [inline bin/metababel:29]: Is this a little more readable?
```
is_regex = pattern.match(%r{^/.+/$})
is_regex ? str =~ eval(pattern) : str == pattern
```
It feed weird to me to extract the thing between  `//` and the using it. Took me will to understand each time

File: bin/metababel (hunk 2 above)
Comment (DonAurelio) [reply bin/metababel:45]: I think it is already there, isn't it?

File: bin/metababel (hunk 3 above)
Comment (DonAurelio) [reply bin/metababel:29]: This looks nice !!. On the other hand, I am wondering if we can inyect ruby code into the pattern, but I didn't find how, so maybe this is safe for now.

Comment (TApplencourt) [reply bin/metababel:29]: if they put `rm -rf *` in the name indeed bad thing will happened... But honestly we don't care

File: bin/metababel (hunk 2 above)
Comment (TApplencourt) [reply bin/metababel:45]: What do you mean?  You want 3 tests `matched_args.uniq.length == match_args.length == matched_args.length`

**Comment by TApplencourt** [comment]
Let's merge! \o/

**Pull request merged**